docs/STEWARDSHIP.md

```diff
@@ -0,0 +1,35 @@
+_NOTE: This document is, and quite likely will always be, a work in progress. You can, and should, contribute to it to make as useful for GitHub Management Stewards as it can be._
+
+# GitHub Management Steward
+
+`github-mgmt stewards` is a team of people who are responsible for managing the GitHub Management configuration for the organization. They have write access to the GitHub Management repository, can review change requests and merge changes to the `master` branch.
+
+Membership in the `github-mgmt stewards` team should be treated with **exactly as much care as having admin access to the organization**.
+
+## What qualifications are expected from a GitHub Management Steward?
+
+- familiarity with [GitHub Management](ABOUT.md)
+- availability for reviews and merges of GitHub Management configuration changes
+- being a trusted member of the organization
+
+## What is expected from a GitHub Management Steward?
+
+- review and merge changes to the GitHub Management configuration
+
+### How to review and merge a GitHub Management pull request?
+
+- Wait for the `Comment` check to pass
+- Verify that the pull request contains only the changes you expect
+- Verify that the plan posted as a comment introduces **only** the changes you expect
+- Check if there are any open PRs created by the `Sync` workflow (titles starting with `sync`) and merge them first if there are
+- Ask the author of the pull request to provide more context if needed
+- Do **not** update the pull request yourself - let the `Update` workflow do it
```

Comment:
I don't understand this bit, @galargh can you provide an example of what this might involve?

Reply:
Sure, let me explain. 

Here's a rough overview of what happens when you have 2 PRs open in `github-mgmt`.

```mermaid
sequenceDiagram
    participant pr1 as PR/1
    participant pr2 as PR/2
    participant branch as Base Branch
    participant gh as GitHub (API)
    participant tf as Terraform (AWS)

    Note over pr1, pr2: The PRs modifying config YAML are opened.
    Note over tf: Current state is v1.
    
    par Plan
        pr1->>tf: What's the current state?
        tf-->>pr1: It's v1!
        Note over pr1: A plan for PR/1 based on v1 is created.<br/>PR/1 can be merged to the base branch.
    and Plan
        pr2->>tf: What's the current state?
        tf-->>pr2: It's v1!
        Note over pr2: A plan for PR/2 based on v1 is created.<br/>PR/2 can be merged to the base branch.
    end
    
    pr1->>branch: Merge!
    
    activate pr2
    
    Note over pr2: PR/2 is not up to date with the base branch anymore.<br/>PR/2 cannot be merged to the base branch.
    
    critical Apply
        branch->>tf: What's the current state?
        tf-->>branch: It's v1!
        Note over branch: This is the same as when the plan for PR/1 was created.
        branch->>gh: Please, apply these changes.
        branch->>tf: Please, update state.
        Note over tf: Current state is v2.
    option Update
        branch->>pr2: Please, update branch.
        Note over pr2: PR/2 is up to date with the base branch again.<br/>PR/2 cannot be merged to the base branch yet<br/>because it does not have a plan yet.
    option Plan
        pr2->>tf: What's the current state?
        tf-->>pr2: It's v2!
        Note over pr2: A plan for PR/2 based on v2 is created.<br/>PR/2 can be merged to the base branch.
    end
    
    deactivate pr2
```

When PR/1 is merged, PR/2 becomes unmergeable because we **require branches to be up to date before merging**. The problem is that as soon as PR/1 is merged, `Update branch` button appears on PR/2. If you click it before the `Apply` workflow run triggered by the PR/1 merge is finished, you risk that the new plan created for PR/2 is going to be based off of tf state v1 instead of tf state v2. If that's the case and you merge that PR/2, the `Apply` workflow run triggered by the PR/2 merge is going to fail because there is going to be tf state mismatch.

One way to solve it would be with [merge queues](https://github.blog/changelog/2021-10-27-pull-request-merge-queue-limited-beta/) but they are still in limited beta and they didn't get back to me on my sign up requests :(

Having written it all down just now though, I got an idea. I think we could resolve this issue if we waited for all the in-progress `Apply` workflow runs to finish before we proceed with actually creating a plan. This should be a pretty straight-forward change and we'd have one less thing to think about when dealing with PRs here. I should be able to get it ready today - will hold off with merging this until then and I'll remove the note afterwards.

Reply:
The change ensuring that we wait for Apply workflow runs to finish before we start creating plans was introduced https://github.com/ipld/github-mgmt/pull/25. Because of this, we can remove this line from the doc because clicking `Update branch` on PRs manually should work just fine now. Of course, it is still not required and all the PRs will keep getting automating updates if they are not up to date yet.
```suggestion
```